pkg/resource/cluster.go

```diff
@@ -105,6 +144,90 @@ func (cluster Cluster) StatefulSetName() string {
 	return cluster.Name()
 }
 
+func (cluster Cluster) JobName() string {
+	return cluster.Name()
+}
+func (cluster Cluster) IsSupportedImage() bool {
+	image := cluster.GetCockroachDBImageName()
+	return !strings.EqualFold(image, NotSupportedVersion)
+}
+func (cluster Cluster) LookupSupportedVersion(version string) (string, bool) {
+	if version == "" {
+		return "", false
+	}
+	supportedVersions := getSupportedCrdbVersions()
+	for _, v := range supportedVersions {
+		if strings.EqualFold(version, v) {
+			return v, true
+		}
+	}
+	return "", false
+}
+
+func (cluster Cluster) GetAnnotationVersion() string {
```

Comment:
nit: `GetVersionAnnotation` reads better.

A comment would help a lot here. It's unclear if this is the version of an annotation or an annotation that represents the version of something.

Reply:
yeap